doc/src/modules/solvers/solveset.rst

```diff
@@ -1,12 +1,463 @@
+
+ * Finitely many solutions (by FiniteSet)
+
+ * Infinitely many solutions, both countably and uncountably infinite solutions
+   (using the ImageSet module) Interval
+
+ * There can also be bizarre solutions to equations like set of all rational number.
+
+No other programmer's object (list, dictionary, python sets) provides the
+flexibility of mathematical sets which our sets module try to emulate. The
+second reason to use sets is that they are close to the entities which
+mathematician's deals with and it makes it easier to reason about them.
+Another advantage of using objects closer to mathematical entities is that the
+user won't have to "learn" our representation and she can have her expectations
+transferred from her mathematical experience.
+
+For multivariate case we are representing solutions as a set of points in a
+n-dimensional space and a point is represented by a FiniteSet of ordered tuple.
+Please Note that, the General FiniteSet is unordered, but a FiniteSet with
+a tuple as it's first argument is ordered. For example:
+
+ - ``FiniteSet(1, 2, 3) : {1, 2, 3}``   # Unordered
+ - ``FiniteSet((1, 2, 3)) {(1, 2, 3)}``  # Ordered
+
+
+Why not use Dictionary as Output?
+
+  Dictionary are easy to deal with programatically but mathematically they are
+  not very precise and use of them can quickly lead to inconsistency and a lot
+  of confusion. For example addition of two solutions is a fairly valid
+  operation but dictionaries by themselves have no add operation. Since we are
+  representing solutions as dictionary we can define a custom add operation
+  where value of `x` variable is added with value of other `x` variable and
+  value of `y` variable with the value of other `y` variable. But what if the user
+  tries to add solution of two equations which have different sets of variables?
+  Say one has `x` and `y` as variables and other has `u` and `v`? We cannot allow this
+  addition because the keys of dictionaries are unordered but this behavior will
+  be inconsistent with the one variable case because it will be absurd to not to
+  allow addition of two numbers, say one obtained by solution of `x = 1` and
+  other by solving `y = 2`.
+
+
+
+Input API of `Solveset`
+-----------------------
+
+``solveset`` has a cleaner input API, unlike ``solve``, It takes a maximum
+of three arguments:
+
+* Equation(s)
+  The equation(s) to solve.
+
+* Variable(s)
+  The variable(s) for which the equation is to be solved.
+
+* domain
+  The domain in which the equation is to be solved.
+
+
+ ``solveset`` removes the ``flags`` argument of ``solve``, which had made
+ the input API messy and output API inconsistent in ``solve``.
+
+
+What is this domain argument about?
+-----------------------------------
+
+ Solveset is designed to be independent of the assumptions on the
+ variable being solved for and instead, uses the ``domain`` argument to
+ decide the solver to dispatch the equation to, namely ``solveset\_real``
```

Comment:
Yeah, perhaps we don't need backslash, if we are using double backticks, but for single backtick we do, as far as I remember.